geo/src/algorithm/buffer.rs

```diff
@@ -0,0 +1,862 @@
+        let default_style = BufferStyle::new(distance);
+        self.buffer_with_style(default_style)
+    }
+
+    /// Create a new geometry whose boundary is offset the specified distance from the input using
+    /// the specific styling options where lines intersect (line joins) and end (end caps).
+    /// For default (rounded) styling, see [`buffer`](Self::buffer).
+    ///
+    /// This method allows control over the buffer appearance through the `BufferStyle` parameter.
+    ///
+    /// # Arguments
+    ///
+    /// * `style` - A `BufferStyle` that specifies the distance, line caps, and line joins.
+    ///
+    /// # Returns
+    ///
+    /// A `MultiPolygon` representing the buffered geometry.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo::{wkt, Buffer};
+    /// use geo::algorithm::buffer::{BufferStyle, LineCap, LineJoin};
+    ///
+    /// let line_string = wkt! { LINESTRING (0. 0.,2. 0.,1. 2.) };
+    /// let style = BufferStyle::new(1.5)
+    ///     .line_cap(LineCap::Square)
+    ///     .line_join(LineJoin::Bevel);
+    /// let buffered = line_string.buffer_with_style(style);
+    /// ```
+    fn buffer_with_style(&self, style: BufferStyle<Self::Scalar>) -> MultiPolygon<Self::Scalar>;
+}
+
+/// Configuration for buffer styling operations.
+///
+/// `BufferStyle` controls how the buffer operation creates the resulting polygon,
+/// including the distance from the original geometry and how line segments are joined or capped.
+///
+/// # Examples
+///
+/// ```
+/// use geo::algorithm::buffer::{BufferStyle, LineCap, LineJoin};
+///
+/// // Default round style
+/// let round_style = BufferStyle::new(2.0);
+///
+/// // Square caps with mitered joins
+/// let square_style = BufferStyle::new(2.0)
+///     .line_cap(LineCap::Square)
+///     .line_join(LineJoin::Miter(1.0));
+/// ```
+pub struct BufferStyle<T: BoolOpsNum> {
+    distance: T,
+    line_cap: LineCap<BoolOpsCoord<T>, T>,
+    line_join: LineJoin<T>,
+}
+
+impl<T: BoolOpsNum> Clone for BufferStyle<T> {
+    fn clone(&self) -> Self {
+        Self {
+            distance: self.distance,
+            line_cap: self.clone_line_cap(),
+            line_join: self.clone_line_join(),
+        }
+    }
+}
+
+impl<T: BoolOpsNum> BufferStyle<T> {
+    /// Creates a new `BufferStyle` with the specified distance and default round caps/joins.
+    ///
+    /// # Arguments
+    ///
+    /// * `distance` - The buffer distance from the original geometry
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo::algorithm::buffer::BufferStyle;
+    ///
+    /// let style = BufferStyle::new(1.5);
+    /// ```
+    pub fn new(distance: T) -> Self {
+        BufferStyle {
+            distance,
+            line_cap: LineCap::Round(Self::default_join_angle()),
+            line_join: LineJoin::Round(Self::default_join_angle()),
+        }
+    }
+
+    /// Sets how two edges of a geometry should meet at vertices.
```

Comment:
Maybe we could state that it doesn't affect `Point`/`MultiPoint`? (cf. also the documentation of `line_join` which refers to cases where the option has no effect)